### PR TITLE
Added Domain Name Reservation Considerations section

### LINKED ISSUE #80: Review by Michelle Thangtamsatid (IANA)
The IANA Functions Operator has a question about one of the actions requested in the IANA Considerations section of this document.

The IANA Functions Operator understands that, upon approval of this document, there are two actions which we must complete.

First, IANA understands that this document calls for the addition of "resolver.arpa" to the Special-Use Domain Names (SUDN) registry established by RFC6761. RFC6761 specifically requires that any request for a Special-Use Domain Name needs to contain a subsection of the "IANA Considerations" section titled Domain Name Reservation Considerations. The current draft does not have such a section.

IANA Question --> Could the authors either conform to the requirements of RFC6761 or provide explicit guidance as to why that section is not appropriate?

Second, in the Transport-Independent Locally-Served DNS Zone Registry on the Locally-Served DNS Zones registry page located at:

https://www.iana.org/assignments/locally-served-dns-zones/locally-served-dns-zones.xhtml

a new registration will be made as follows:

Zone: resolver.arpa.
Description: DNS Resolver Special-Use Domain
Reference: [ RFC-to-be ]

The IANA Functions Operator understands that these are the only actions required to be completed upon approval of this document.

Note:  The actions requested in this document will not be completed until the document has been approved for publication as an RFC. This message is meant only to confirm the list of actions that will be performed.

For definitions of IANA review states, please see:

https://datatracker.ietf.org/help/state/draft/iana-review

### PR DESCRIPTION
**Pull request by mstojens**
Resolves #80

### CODE REVIEW

File: draft-ietf-add-ddr.md [hunk 1]
```diff
@@ -523,6 +523,66 @@ IANA is requested to add an entry in "Transport-Independent Locally-Served
...
+
+1. Are human users expected to recognize these names as special and use them
+differently? In what way?
+
+No. This name should be used automatically by DNS clients on behalf of users.
+
+2.  Are writers of application software expected to make their software
+recognize these names as special and treat them differently? In what way?
+
+No. There is no use case where a non-DNS application (covered by the next
```
Comment (tfpauly) [inline draft-ietf-add-ddr.md:539]:
```suggestion
No. There is no use case where a non-DNS application (covered by the next
```

File: draft-ietf-add-ddr.md [hunk 2]
```diff
@@ -523,6 +523,66 @@ IANA is requested to add an entry in "Transport-Independent Locally-Served
...
+instead of checking the matching zone as it would for normal domain names.
+
+4. Are developers of caching domain name servers expected to make their
+implementations recognize these names as special and treat them differently?
+If so, how?
+
+Yes. Caching domain name servers should not forward queries for this name to
+avoid causing validation failures due to IP address mismatch.
+
+5. Are developers of authoritative domain name servers expected to make their
```
Comment (tfpauly) [inline draft-ietf-add-ddr.md:557]:
```suggestion
5. Are developers of authoritative domain name servers expected to make their
```

File: draft-ietf-add-ddr.md [hunk 3]
```diff
@@ -523,6 +523,66 @@ IANA is requested to add an entry in "Transport-Independent Locally-Served
...
+as authoritative for this reserved name, will compliant name server software
+reject it as invalid? Do DNS server operators need to know about that and
+understand why? Even if the name server software doesn't prevent them from
+using this reserved name, are there other ways that it may not work as expected,
+of which the DNS server operator should be aware?
+
+This name is locally served, and any resolver which supports this name should
+never forward the query. DNS server operators should be aware that records for
+this name will be used by clients to modify the way they connect to their
+resolvers.
```
Comment (tfpauly) [inline draft-ietf-add-ddr.md:577]:
```suggestion
resolvers.
```

File: draft-ietf-add-ddr.md [hunk 4]
```diff
@@ -523,6 +523,66 @@ IANA is requested to add an entry in "Transport-Independent Locally-Served
 DNS Zones" registry for 'resolver.arpa.' with the description "DNS Resolver
 Special-Use Domain", listing this document as the reference.
 
+## Domain Name Reservation Considerations
+
+In accordance with section 8.1 of {{?RFC6761}}, the answers to the following
```
Comment (tfpauly) [inline draft-ietf-add-ddr.md:528]:
```suggestion
In accordance with {{Section 5 of RFC6761}}, the answers to the following
```

File: draft-ietf-add-ddr.md [hunk 5]
```diff
@@ -523,6 +523,66 @@ IANA is requested to add an entry in "Transport-Independent Locally-Served
...
+questions are provided relative to this document:
+
+1. Are human users expected to recognize these names as special and use them
+differently? In what way?
+
+No. This name should be used automatically by DNS clients on behalf of users.
```
Comment (tfpauly) [inline draft-ietf-add-ddr.md:534]:
```suggestion
No. This name is used automatically by DNS stub resolvers running on client devices on behalf of users, and users will never see this name directly.
```

File: draft-ietf-add-ddr.md [hunk 6]
```diff
@@ -523,6 +523,66 @@ IANA is requested to add an entry in "Transport-Independent Locally-Served
...
+
+2.  Are writers of application software expected to make their software
```
Comment (tfpauly) [inline draft-ietf-add-ddr.md:536]:
```suggestion
2. Are writers of application software expected to make their software
```

File: draft-ietf-add-ddr.md [hunk 7]
```diff
@@ -523,6 +523,66 @@ IANA is requested to add an entry in "Transport-Independent Locally-Served
...
+This name is locally served, and any resolver which supports this name should
+never forward the query. DNS server operators should be aware that records for
+this name will be used by clients to modify the way they connect to their
+resolvers.
+
+7. How should DNS Registries/Registrars treat requests to register this reserved
+domain name? Should such requests be denied? Should such requests be allowed,
+but only to a specially-designated entity?
+
+IANA should hold the registry for this name. Non-IANA requests to register
```
Comment (tfpauly) [inline draft-ietf-add-ddr.md:583]:
```suggestion
IANA should hold the registration for this name. Non-IANA requests to register
```

**Review by tfpauly — CHANGES_REQUESTED** [review]

File: draft-ietf-add-ddr.md (hunk 4 above)
Comment (mstojens) [reply draft-ietf-add-ddr.md:528]: *sigh* well _that_ would explain the failures. Apparently too many edits in one sprint. Thanks for catching this.

**Comment by mstojens** [comment]
I have committed all request changes. Thank you for the refinements.

**Review by tfpauly — APPROVED** [review]

**Pull request merged**